### PR TITLE
Remove wrapPanic

### LINKED ISSUE #411: Remove wrapPanic
the `wrapPanic` function conveniently wraps any panics coming up the stack to assist building a stack trace for error reporting. However, it causes problems when unexpected panics happen due to implementation bugs.

For every panic arising from an error in arr.ai code, change the logic to return a Go `error` instead. Once this is done, remove `wrapPanic`.

Panics representing implementation bugs can remain as is, since they assist arr.ai maintainers to track down such bugs.

### PR DESCRIPTION
**Pull request by marcelocantos**
Fixes #411 .

Changes proposed in this pull request:
- Add error return to several functions and interface methods
- Replace many panics with errors
- Fix all test scenarios that depend on wrapPanic.
- Remove wrapPanic

Checklist:
- [ ] Added related tests
- [ ] Made corresponding changes to the documentation

### CODE REVIEW

File: rel/pattern_ident.go [hunk 1]
```diff
@@ -9,8 +11,12 @@ func NewIdentPattern(ident string) IdentPattern {
 }
 
 func (p IdentPattern) Bind(scope Scope, value Value) (Scope, error) {
-	scope.MustGet(p.ident)
-	scope.MatchedWith(p.ident, value)
+	if _, has := scope.Get(p.ident); !has {
+		return EmptyScope, fmt.Errorf("%q not in scope", p.ident)
```
Comment (nofun97) [inline rel/pattern_ident.go:15]: should this use `wrapError`? I assume `IdentPattern` is an expression

File: rel/pattern_set.go [hunk 2]
```diff
@@ -74,11 +74,14 @@ func (p SetPattern) Bind(local Scope, value Value) (Scope, error) {
 				return EmptyScope, fmt.Errorf("item type %s is not supported yet", t)
 			}
 		case IdentPattern:
-			v := local.MustGet(t.ident).(Value)
-			if !set.Has(v) {
+			v, has := local.Get(t.ident)
+			if !has {
+				return EmptyScope, fmt.Errorf("%q not in scope", t.ident)
```
Comment (nofun97) [inline rel/pattern_set.go:79]: ditto

File: syntax/std_re.go [hunk 3]
```diff
@@ -1,17 +1,25 @@
...
 
 var (
-	stdReMatch = rel.NewNativeFunction("compile", func(re rel.Value) rel.Value {
-		regex := regexp.MustCompile(mustAsString(re))
+	stdReMatch = rel.NewNativeFunction("compile", func(re rel.Value) (rel.Value, error) {
+		reStr, is := valueAsString(re)
+		if !is {
+			return nil, fmt.Errorf("//re.compile: re not a string: %v", re)
+		}
+		regex := regexp.MustCompile(reStr)
```
Comment (nofun97) [inline syntax/std_re.go:16]: return error

File: syntax/std_seq_array_helper.go [hunk 4]
```diff
@@ -108,22 +127,25 @@ func arrayHasPrefix(prefix rel.Value, subject rel.Array) rel.Value {
...
+	return rel.NewBool(true), nil
 }
 
 // Check if array subject ends with suffix.
-func arrayHasSuffix(suffix rel.Value, subject rel.Array) rel.Value {
-	suffixArray := convert2Array(suffix)
+func arrayHasSuffix(suffix rel.Value, subject rel.Array) (rel.Value, error) {
+	suffixArray, is := rel.AsArray(suffix)
+	if !is {
+		return nil, fmt.Errorf("has_suffix: suffix not an array: %v", suffix)
```
Comment (nofun97) [inline syntax/std_seq_array_helper.go:141]: error message isn't very consistent, some only show the function name others show the full name.

File: syntax/std_str.go [hunk 5]
```diff
@@ -53,48 +60,67 @@ var (
 				}
 				s = sb.String()
 			} else {
-				panic(fmt.Errorf("arg not an array in ${arg::}: %v", args[1]))
+				return nil, fmt.Errorf("arg not an array in ${arg::}: %v", args[1])
```
Comment (nofun97) [inline syntax/std_str.go:63]: mention fucntion name

File: syntax/std_re.go (hunk 3 above)
Comment (marcelocantos) [reply syntax/std_re.go]: Done.

File: syntax/std_seq_array_helper.go (hunk 4 above)
Comment (marcelocantos) [reply syntax/std_seq_array_helper.go]: Done.

File: syntax/std_str.go (hunk 5 above)
Comment (marcelocantos) [reply syntax/std_str.go]: Done.

**Pull request merged**